src/main/scala/strawman/collection/View.scala

```diff
@@ -286,11 +272,11 @@ object IndexedView {
     def apply(i: Int) = underlying.apply(i)
   }
 
-  class TakeRight[A](underlying: IndexedView[A], n: Int)
-  extends View.TakeRight(underlying, n) with IndexedView[A] {
-    override def iterator() = super.iterator() // needed to avoid "conflicting overrides" error
+  class TakeRight[A](underlying: IndexedView[A], n: Int) extends IndexedView[A] {
+    private val normN = n max 0
+    private val delta = underlying.length - normN
     def length = underlying.length min normN
-    def apply(i: Int) = underlying.apply(i)
+    def apply(i: Int) = underlying.apply(i + delta)
```

Comment:
I kept the `IndexedView` based implementation of `takeRight` (and `dropRight`, below)